sc-tools/sc-machine-runner/src/sc_machine_setup.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+ * Distributed under the MIT License
+ * (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+ */
+
+#include "sc_machine_setup.hpp"
+
+#include <iostream>
+#include <atomic>
+#include <thread>
+
+#include <sc-memory/sc_memory.hpp>
+#include <sc-memory/sc_debug.hpp>
+#include <sc-memory/utils/sc_signal_handler.hpp>
+
+#include <sc_options.hpp>
+#include <sc_memory_config.hpp>
+
+void PrintStartMessage()
+{
+  std::cout << "SC-MACHINE USAGE\n\n"
+            << "--config|-c -- Path to configuration .ini file\n"
+            << "--extensions_path|-e -- Path to directory with sc-memory extensions\n"
+            << "--repo_path|-r -- Path to kb.bin directory\n"
+            << "--clear -- Flag to clear sc-memory state on initialize\n"
+            << "--verbose|-v -- Flag to don't save sc-memory state on shutdown\n"
+            << "--test|-t -- Flag to test sc-machine, run and stop it\n"
+            << "--help -- Display this message\n\n";
+}
+
+sc_int BuildAndRunMachine(sc_int argc, sc_char * argv[])
+try
+{
+  ScOptions options{argc, argv};
+
+  if (options.Has({"help"}))
+  {
+    PrintStartMessage();
+    return EXIT_SUCCESS;
+  }
+
+  std::string configFile;
+  if (options.Has({"config", "c"}))
+    configFile = options[{"config", "c"}].second;
+
+  sc_bool saveOnShutdown = !options.Has({"verbose", "v"});
+
+  ScMemory::ms_configPath = configFile;
+
+  ScConfig config{configFile, {"repo_path", "extensions_path", "log_file"}};
+  ScParams memoryParams{options, {{"extensions_path", "e"}, {"repo_path", "r"}, {"clear"}}};
+  ScMemoryConfig memoryConfig{config, memoryParams};
+
+  std::atomic_bool isRun;
```

Comment:
Can you move this line before it is used (isRun = !options.Has({"test", "t"});)

Reply:
No, because there can be unitialized value due jump by goto